datastore/src/main/java/com/ds/datastore/BookController.java

```diff
@@ -0,0 +1,79 @@
+package com.ds.datastore;
+
+
+import org.springframework.hateoas.CollectionModel;
+import org.springframework.hateoas.EntityModel;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static org.springframework.hateoas.server.mvc.WebMvcLinkBuilder.linkTo;
+import static org.springframework.hateoas.server.mvc.WebMvcLinkBuilder.methodOn;
+
+@RestController
+public class BookController {
+
+    private final BookRepository repository;
+
+    private final BookModelAssembler assembler;
+
+    BookController(BookRepository repository, BookModelAssembler assembler) {
+
+        this.repository = repository;
+        this.assembler = assembler;
+    }
+
+    @PostMapping("/books")
+    Book newBook(@RequestBody Book book) {
+        return repository.save(book);
```

Comment:
By default, controllers return status code `200`. 

The convention is that a creation should return a `201` - so let's return that status code here.

Bonus points - the convention is also to return a link to the newly created object in the `Location` header. Lets do that as well. Let me know if you can't figure it out.